contracts/strategies/DirectAllocation.sol

```diff
@@ -69,15 +74,23 @@ contract DirectAllocationStrategy is CoreBaseStrategy {
             revert INVALID_INPUT();
         }
 
+        uint256 _totalNativeAmount;
         for (uint256 _i = 0; _i < _recipientsLength;) {
             /// Direct allocate the funds
-            _transferAmountFrom(_tokens[_i], TransferData({from: _sender, to: _recipients[_i], amount: _amounts[_i]}));
+            if (_tokens[_i] == NATIVE) {
+                _totalNativeAmount += _amounts[_i];
+                _recipients[_i].safeTransferETH(_amounts[_i]);
+            } else {
+                _tokens[_i].safeTransferFrom(_sender, _recipients[_i], _amounts[_i]);
+            }
```

Comment:
Why not use the `transferAmount` here and only have an if for accounting the `_totalNativeAmount` ?

Reply:
`transferAmount` uses `safeTransfer` not `safeTransferFrom`. We could add a function like the following:

```solidity
    function transferAmountFrom(address _token, address _from, address _to, uint256 _amount) internal {
        if (_token == NATIVE) {
            // '_from' is ignored. The contract's balance is used. 
            if (_to != address(this)) _to.safeTransferETH(_amount);
        } else {
            _token.safeTransferFrom(_from, _to, _amount);
        }
    }
```

It would be slightly inefficient and the dev won't be completely abstracted away from what's happening (native balance still needs to be checked), but `SafeTransferLib` won't be needed and the code might look simpler in some cases.

I was reluctant to keep transferAmountFrom, but thinking again about it I like it. Wdyt?

Reply:
I think based on the use cases we currently have it makes sense to have the `transferAmountFrom`, probably worth adding it back